### PR TITLE
Index malleated transactions properly

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

Simple PR to utilize the malleated/original transaction scheme used in #582 and #607 to index malleated transaction by the hash of the original transaction. This allows for querying the transaction using the original hash, instead of the hash of the transaction that actually ends up on chain

### CODE REVIEW

**Review by SweeXordious — APPROVED** [review]

**Comment by SweeXordious** [comment]
@evan-forbes Should we build a new node depending on this change and test the query ?

**Comment by evan-forbes** [comment]
> Should we build a new node depending on this change and test the query?

yeah, we can. I should also look for a more integrated way to test before merging tbh

**Comment by evan-forbes** [comment]
After writing an integration test in the app, I found out that without further modification, the method of unwrapping the tx to get the original tx won't work.

still working on a different solution atm, so I'm marking this as a draft as it should not yet be merged

File: proto/tendermint/abci/types.proto [hunk 1]
```diff
@@ -338,6 +338,7 @@ message TxResult {
   uint32            index  = 2;
   bytes             tx     = 3;
   ResponseDeliverTx result = 4 [(gogoproto.nullable) = false];
+  bytes original_hash = 5;
```
Comment (evan-forbes) [inline proto/tendermint/abci/types.proto:341]: I modified abci.TxResult to include the original hash if the tx is malleated. This saves us from having to attempt to unwrap the malleated tx more than once. Alternatively, we could not cache the original hash, and unwrap the malleated transaction more than once.

**Review by evan-forbes — COMMENTED** [review]
alrighty, got it working faster than I thought I would. There's an integration test in the app that submits a `WirePayForMessage` and then attempts to query for the transaction using the original tx hash. 

[#188](https://github.com/celestiaorg/celestia-app/pull/188)

File: proto/tendermint/abci/types.proto (hunk 1 above)
Comment (adlerjohn) [inline proto/tendermint/abci/types.proto:341]: Can you align this field like the others?

**Review by adlerjohn — COMMENTED** [review]
Is this test app CI supposed to fail?

**Comment by evan-forbes** [comment]
> Is this test app CI supposed to fail?

looking into this

File: abci/types/application.go [hunk 2]
```diff
@@ -98,7 +98,7 @@ func (BaseApplication) ApplySnapshotChunk(req RequestApplySnapshotChunk) Respons
 }
 
 func (BaseApplication) PreprocessTxs(req RequestPreprocessTxs) ResponsePreprocessTxs {
-	return ResponsePreprocessTxs{}
+	return ResponsePreprocessTxs{Txs: req.Txs}
```
Comment (evan-forbes) [inline abci/types/application.go:101]: this took entirely too long to debug... not that it matters too much cause upstream removed these tests in [#6685](https://github.com/tendermint/tendermint/pull/6684)

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**